front/lib/api/assistant/generation.ts

```diff
@@ -106,6 +106,16 @@ export async function renderConversationForModel({
         name: m.context.username,
         content,
       });
+    } else if (isContentFragmentType(m)) {
+      messages.unshift({
+        role: "content_fragment" as const,
+        name: "inject_content_fragment",
+        content: m.content,
```

Comment:
I would do something a bit more elaborated assuming you add a title:
```
TITLE: ${title}
CONTENT:
${content}
```

Reply:
Ok gotcha